src/maps/components/colorbar-plot.tsx

```diff
@@ -1,60 +1,138 @@
-      return (
-        <table key={marker.position.index} className="colorbar-table">
-          <thead>
-            <tr>
-              <th colSpan={2}>
-                {label}
-              </th>
-            </tr>
-          </thead>
-          <tbody>
-          {[...ymdDates].reverse().map((date) => {
-            const value = observations[date][position.index];
-            const color = getColorForValue(value, minValue, maxValue);
-            return (
-              <tr key={date}>
-                <td
-                  style={{
-                    width: "40px",
-                    fontWeight: date === selectedYMDDate ? "bold" : "normal",
-                    backgroundColor: date === selectedYMDDate ? "lightgray" : "white"
-                  }}
-                  >{abbrDate(new Date(date))}</td>
-                <td
-                  title={`${value}`}
-                  style={{backgroundColor: color, width: "60px"}}
-                />
-              </tr>
-            )
-          })}
-          </tbody>
-        </table>
-      )
-    });
+  const { observations, ymdDates, placenames, minValue, maxValue, range } = dataSet;
+  const labels = selectedMarkers.map((marker) => placename(marker.position, placenames))
+
+  const yLabels = ymdDates.map((ymd) => formatDate(new Date(ymd)));
+  // get next month and add to yLabels
+  const nextMonth = new Date(ymdDates[ymdDates.length - 1]).getMonth() + 1;
+  const year = new Date(ymdDates[ymdDates.length - 1]).getFullYear();
+  const nextMonthDate = formatDate(new Date(year, nextMonth));
+  yLabels.push(nextMonthDate);
+
+  const options = {
+    plugins: {
+      title: {
+        display: false,
+        text: '',
+      },
+      legend: {
+        display: false,
+      },
+      tooltip: {
+        callbacks: {
+          title: function(context: any) {
+            const date = context[0].raw.y;
+            return date[0];
+          },
+          label: function(context: any) {
+            const ymdDate = ymdDates[context.parsed.y - 1];
+            const location = selectedMarkers[context.parsed.x].position;
+            const observation = observations[ymdDate][location.index];
+            return `${placename(location, placenames)}: ${observation}`;
+          }
+        }
+
+      },
+      annotation: {
+        annotations: {
+          box1: {
+            type: "box" as const,
+            yMin: ymdDates.indexOf(selectedYMDDate!),
+            yMax: ymdDates.indexOf(selectedYMDDate!) + 1,
+            backgroundColor: "rgba(255, 99, 132, 0.25)"
+          }
+        }
+      }
+    },
+    responsive: true,
+    scales: {
+      x: {
+        stacked: true,
+        type: 'category' as const,
+        barPercentage: .5,
+      },
+      y: {
+        reverse: true,
+        type: 'category' as const,
+        labels: yLabels,
+      }
+    },
+  };
+
+  const datasets: Array<IDataset> = [];
+  const bins: Array<number> = [];
+  const binSize = range / 20;
```

Comment:
Maybe make `20` a constant here and update the next couple of lines to use the constant instead of the literal `19` and `20`?